### ISSUE TITLE
Concurrency Tokens on SQLite in EF Core

### ISSUE THREAD

**Issue by bricelam** [bricelam/bricelam.github.io#19]
https://github.com/aspnet/EntityFrameworkCore/issues/12260#issuecomment-395509124

**Comment by ggirard07** [comment]
That's what I am trying to achieve. But in order to do so, I need to inject parameters in the SQL command, which always end-up with `SQLite Error 1: 'near "@p0": syntax error'`.

I tried to parameterized the command as follow:
```csharp
db.Database.ExecuteSqlCommand($@"
    CREATE TRIGGER PatchRowVersionInsert{entity.TableName}
    AFTER INSERT ON {entity.TableName}
    BEGIN
        UPDATE {entity.TableName}
        SET {entity.RowVersionPropertyName} = randomblob(8)
        WHERE {string.Join(" AND ", entity.PrimaryKeyNames.Select(n => $"{n} = NEW.{n}"))};
    END
");
```
I tried to hardcode some of the variable in case parameters were not supported in the trigger name for example, but exception always happens on `@p0`.
`EnsureCreated()` has already been called at that time and some seed has already happened in some tables.

I am using an in-memory SQLite database with ASP.Net Core 2.1 configured with [implicit version](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/metapackage-app?view=aspnetcore-2.2).

**Comment by bricelam** [comment]
You’re hitting a quirk of ExecuteSqlComnand. Cast the string to a string:
``` CSharp
db.Databae.ExecuteSqlCommand((string)$@“...”);
```

**Comment by ggirard07** [comment]
Casting to string ends up in the same workaround I am currently using (expanding the string first then pass it to `ExecuteSqlCommand`), which result in the SQL injection warning.

Is there any proper fix for this, except than disabling the warning?

**Comment by bricelam** [comment]
We fixed it in EF Core 3.0
